src/VocabularyManager.Api/Middleware/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Net;
+using VocabularyManager.Api.Models.Responses;
+using VocabularyManager.UseCases.Exceptions;
+
+namespace VocabularyManager.Api.Middleware
+{
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly ILogger<ExceptionHandlingMiddleware> _logger;
+        private readonly RequestDelegate _nextMiddleware;
+
+        public ExceptionHandlingMiddleware(
+            ILogger<ExceptionHandlingMiddleware> logger,
+            RequestDelegate nextMiddleware
+        )
+        {
+            _logger = logger;
+            _nextMiddleware = nextMiddleware;
+        }
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            try
+            {
+                await _nextMiddleware(httpContext);
+            }
+            catch(Exception e)
+            {
+                await HandleExceptinAsync(httpContext, e);
+            }
+        }
+        private async Task HandleExceptinAsync(HttpContext httpContext, Exception e)
+        {
+            _logger.LogError($"{DateTime.UtcNow} - excpetion handled: {e.Message}");
+            HttpStatusCode statusCode = DefineStatusCode(e);
+            httpContext.Response.StatusCode = (int) statusCode;
+            await httpContext.Response.WriteAsJsonAsync(
+                new ExceptionResponse(statusCode, e.Message)
+                );
+        }
+        private HttpStatusCode DefineStatusCode(Exception e)
+        {
+            switch(e)
+            {
+                case WordNotFoundException:
+                case VocabularyNotFoundException:
+                    return HttpStatusCode.NotFound;
```

Comment:
NotFound it's used for endpoints that does not exist. Not for resources that are not found.

You can return 400 Bad Request error with detail message that `something was not found`, Descriptive errors for consumer it's a big deal
https://softwareengineering.stackexchange.com/questions/203492/when-to-use-http-status-code-404-in-an-api